Source/Core/VideoBackends/Vulkan/VulkanContext.cpp

```diff
@@ -584,8 +611,12 @@ bool VulkanContext::CreateDevice(VkSurfaceKHR surface, bool enable_validation_la
   if (!LoadVulkanDeviceFunctions(m_device))
     return false;
 
-  // Grab the graphics queue (only one we're using at this point).
+  // Grab the graphics and present queues.
   vkGetDeviceQueue(m_device, m_graphics_queue_family_index, 0, &m_graphics_queue);
+  if (surface)
+  {
+    vkGetDeviceQueue(m_device, m_present_queue_family_index, 0, &m_present_queue);
```

Comment:
Does the spec guarantee that the same handle will be returned twice when called with the same queue family index?

Wondering if something like this would be better:
if (m_graphics_queue_family_index != m_present_queue_family_index)
  vkGetDeviceQueue(m_device, m_present_queue_family_index, 0, &m_present_queue);
else
  m_present_queue = m_graphics_queue;

Reply:
If they are the same underlying queue family then we set `device_info.queueCreateInfoCount = 1`, so we only actually make one underlying queue. Then we retrieve it twice because the third argument to `vkGetDeviceQueue`, `queueIndex`, is 0 for both.

Reply:
Okay, should be fine then.